packages/mint-components/src/components/sqm-task-card/sqm-task-card.tsx

```
@@ -0,0 +1,216 @@
+
+  /**
+   * @uiName Goal Completion Number
+   */
+  @Prop()
+  goal: number = 1;
+
+  /**
+   * @uiName Progress Bar Steps
+   */
+  @Prop()
+  steps: boolean = false;
+
+  /**
+   * @uiName Progress Bar Unit
+   */
+  @Prop()
+  progressBarUnit: string = "";
+
+  /**
+   * @uiName Expiry Date Message
+   */
+  @Prop() expiryMessage: string = "Ends {endDate}";
+
+  /**
+   * @uiName Reward Duration
+   * @uiWidget DateRange
+   * @uiOptions {"allowPastDates":true, "months": 1}
+   */
+  @Prop() rewardDuration: string;
+
+  /**
+   * Shown to users before the start of the task duration.
+   *
+   * @uiName Start Date Message
+   */
+  @Prop() startsOnMessage: string = "Starts {startDate}";
+
+  /**
+   * Timeframe that the task card will be shown
+   *
+   * @uiName Display Duration
+   * @uiWidget DateRange
+   * @uiOptions {"allowPastDates":true, "months": 1}
+   */
+  @Prop() displayDuration: string;
+
+  /**
+   * Shown to users after the end of the task duration.
+   *
+   * @uiName Ended Date Message
+   */
+  @Prop() endedMessage: string = "Ended {endDate}";
+
+  /**
+   * Displays the amount of times that a user has completed a repeatable task.
+   *
+   * @uiName Completed Count Text
+   */
+  @Prop() completedText: string =
+    "Completed {finite, select, 0 {{count, plural, =1 {{count} time} other {{count} times}}} other {{count}/{finite} times}}";
+
+  /**
+   * @uiName CTA Button Text
+   */
+  @Prop()
+  buttonText: string = "Complete Action";
+
+  /**
+   * @uiName CTA Button Link
+   */
+  @Prop()
+  buttonLink: string = "https://example.com/";
+
+  /**
+   * @uiName CTA Button Link Open in New Tab
+   */
+  @Prop()
+  openNewTab: boolean = false;
+
+  /**
+   * Select what type of stat to display for the goal. Manual paths are also supported.
+   *
+   * @uiWidget StatTypeSelectWidget
+   * @uiName Goal Progress Source
+   * @uiOptions {"version": 1.1}
+   */
+  @Prop()
+  statType: string = "/programGoals/count/Referral-Started%2Freferrals";
```

@00salmon I am thinking we might actually want to remove this default, thoughts?

---

I'm pretty indifferent, that goal is one that will exist by default for anyone using a referral program i think, but if they are loyalty only it wouldn't be so useful 🤔

---

I'm wondering if we actually want to follow a more fail fast approach

---

we do have a fail-fast component for required props 👀